### PR TITLE
Why3 1.4.0

### PR DESCRIPTION
**Pull request by silene**

### CODE REVIEW

File: packages/why3/why3.1.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  "ocaml" {>= "4.05.0"}
+  "ocamlfind" {build}
+  "menhir" {>= "20151112"}
+  "num"
+]
+
+depopts: [
+  "zarith"
+  "camlzip"
+  "ocamlgraph"
```
Comment (kit-ty-kate) [inline packages/why3/why3.1.4.0/opam:53]: with ocamlgraph installed this package fails with
```
# Linking  lib/plugins/genequlin.cmo
# Linking  lib/plugins/dimacs.cmo
# Linking  bin/why3extract.cma
# Linking  bin/why3wc.cma
# Linking  bin/why3webserver.cma
# Linking  bin/why3shell.cma
# Linking  bin/isabelle_client.byte
# Linking  bin/why3pp.cma
# Ocamlopt src/driver/collect_data_model.ml
# Linking  bin/why3prove.cma
# Linking  lib/plugins/hypothesis_selection.cmo
# File "_none_", line 1:
# Error: File graph.cmo not found
# make: *** [Makefile:587: lib/plugins/hypothesis_selection.cmo] Error 2
# make: *** Waiting for unfinished jobs....
```

**Comment by kit-ty-kate** [comment]
also frama-c fails with:
```
# File "src/plugins/wp/Why3Provers.ml", line 48, characters 32-42:
# 48 |               (Why3.Debug.Args.[desc_debug;desc_debug_all;desc_debug_list])
#                                      ^^^^^^^^^^
# Error: This expression has type
#          Why3.Debug.Args.spec =
#            Why3.Getopt.key * Why3.Getopt.handler * Why3.Getopt.doc
#        but an expression was expected of type Arg.key * Arg.spec * Arg.doc
#        Type Why3.Getopt.key is not compatible with type Arg.key = string 
# make: *** [share/Makefile.generic:78: src/plugins/wp/Why3Provers.cmo] Error 2
# make: *** Waiting for unfinished jobs....
```
is that expected? if yes, i'll fix their constraints

File: packages/why3/why3.1.4.0/opam (hunk 1 above)
Comment (silene) [reply packages/why3/why3.1.4.0/opam:54]: The packaging of ocamlgraph changed between 1.8 and 2.0. I have disabled support for ocamlgraph 2.

**Comment by silene** [comment]
> is that expected? if yes, i'll fix their constraints

That is unfortunate, but I don't see a way around this breakage, so yes.

File: packages/why3/why3.1.4.0/opam (hunk 1 above)
Comment (silene) [reply packages/why3/why3.1.4.0/opam:54]: This now passes CI.

**Comment by silene** [comment]
Do you want me to adjust the package constraints for Frama-C?

**Comment by mseri** [comment]
Yes please!

**Comment by mseri** [comment]
Thank you very much, this was very helpful. Sorry for the not having done it earlier by ourselves

**Comment by mseri** [comment]
This seems good to go. Any eventual further  refinement of frama-c can be dealt with separately

**Pull request merged**